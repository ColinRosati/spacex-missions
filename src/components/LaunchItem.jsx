import React from 'react';
import '../styles/launch.css';

function LaunchItem(props) {
  const {mission, nationality, manufacturer, type} = props
  return (
    <div className="launch-item" onClick={props.onClick}>
      <div className="launch-item-mission">{mission}</div>
      <div className="launch-item-nationality">{nationality}</div>
      <div className="launch-item-manufacturer">{manufacturer}</div>
      <div className="launch-item-type">{type}</div>
    </div>
  );
}

export default LaunchItem;