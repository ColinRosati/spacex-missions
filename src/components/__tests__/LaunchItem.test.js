import LaunchItem from '../LaunchItem';
import React from 'react'
import Enzyme, { shallow, mount } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'

Enzyme.configure({ adapter: new Adapter() }) 

describe('LaunchItem component', () => {
    it('should render hitoryitems component', () => {
        const wrapper = shallow(<LaunchItem/>);
        expect(wrapper).toMatchSnapshot();
    });

    it('should expect props to be valid', () => {
        const it = mount(<LaunchItem/>);
        expect(it.props().mission).not.toBeNull();
        expect(it.props().nationality).not.toBeNull();
        expect(it.props().manufacturer).not.toBeNull();
        expect(it.props().type).not.toBeNull();
    })

    it('should have 4 divs', () => {
        const wrapper = shallow( <LaunchItem/> )
        const children = wrapper.find('div').children()
        expect(children.length).toBe(4);
    })

})