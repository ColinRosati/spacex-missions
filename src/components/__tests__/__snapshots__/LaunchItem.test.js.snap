// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LaunchItem component should render hitoryitems component 1`] = `ShallowWrapper {}`;
